src/code/z_collision_check.c

```diff
@@ -1,392 +1,3345 @@
+                    *arg6 = sp60;
+                    return 1;
+                }
+            }
+            return 0;
+        }
+    }
+    // ada468 800632C8
+    if (phi_v0_3 == 0) { // ada468:    bnezl   v0,0xada4a4 ~>
+        if (sp50 < 0.0f || 1.0f < sp50) {
+            return 0;
+        }
+    } else { // ada4a4
+        phi_a1 = 0;
+        if (sp50 < 0.0f) {
+            phi_a1 = 1;
+        }
+        if (phi_a1 == 0) {
+            phi_a1 = 0;
+            if (1.0f < sp50) {
+                phi_a1 = 1;
+            }
+        }
+        phi_a2 = 0;
+        if (sp4C < 0.0f) {
+            phi_a2 = 1;
+        }
+        if (phi_a2 == 0) {
+            phi_a2 = 0;
+            if (1.0f < sp4C) {
+                phi_a2 = 1;
+            }
+        }
+        if (phi_a1 != 0) {
+            if (phi_a2 != 0) {
+                return 0;
+            }
+        }
+        if (phi_a1 != 0) {
+            phi_v1_2 = 0;
+        }
+        if (phi_a2 != 0) {
+            phi_v0_3 = 0;
+        }
+    }
+    if (phi_v1_2 == 1) {
+        temp_f0_3 = (sp50 * sp54.y) + sp6C.y;
+        if ((temp_f0_3 < 0.0f) || (actor_ac_98_12 < temp_f0_3)) {
+            phi_v1_2 = 0;
+        }
+    }
+    if (phi_v0_3 == 1) {
+        temp_f0_4 = (sp4C * sp54.y) + sp6C.y;
+        if ((temp_f0_4 < 0.0f) || (actor_ac_98_12 < temp_f0_4)) {
+            phi_v0_3 = 0;
+        }
+    }
+    if (phi_v1_2 == 0 && phi_v0_3 == 0) {
+        return 0;
+    }
+    if (phi_v1_2 == 1) {
+        if (phi_v0_3 == 1) {
+            arg6->x = (f32)(((sp50 * sp54.x) + sp6C.x) + arg3->x);
+            arg6->y = (f32)(((sp50 * sp54.y) + sp6C.y) + arg3->y);
+            arg6->z = (f32)(((sp50 * sp54.z) + sp6C.z) + arg3->z);
+            arg7->x = (f32)(((sp4C * sp54.x) + sp6C.x) + arg3->x);
+            arg7->y = (f32)(((sp4C * sp54.y) + sp6C.y) + arg3->y);
+            arg7->z = (f32)(((sp4C * sp54.z) + sp6C.z) + arg3->z);
+            return 2;
+        }
+    }
+    if (phi_v1_2 == 1) {
+        arg6->x = (f32)(((sp50 * sp54.x) + sp6C.x) + arg3->x);
+        arg6->y = (f32)(((sp50 * sp54.y) + sp6C.y) + arg3->y);
+        arg6->z = (f32)(((sp50 * sp54.z) + sp6C.z) + arg3->z);
+        return 1;
+    }
+    if (phi_v0_3 == 1) { // ada700
+        arg6->x = (f32)(((sp4C * sp54.x) + sp6C.x) + arg3->x);
+        arg6->y = (f32)(((sp4C * sp54.y) + sp6C.y) + arg3->y);
+        arg6->z = (f32)(((sp4C * sp54.z) + sp6C.z) + arg3->z);
+        return 1;
+    }
+    return 1;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_80062ECC.s")
+#endif // NON_MATCHING
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_800635D0.s")
+s32 func_800635D0(s32 arg0) {
+    s32 result;
+
+    result = 0;
+    if ((arg0 & 0x00400100) != 0) {
+        result = 1;
+    } else if ((arg0 & 0x03000242) != 0) {
+        result = 2;
+    } else if ((arg0 & 0x48800400) != 0) {
+        result = 4;
+    } else if ((arg0 << 5) < 0) {
+
+        result = 8;
```

Comment:
```suggestion
    if (arg0 & 0x00400100) {
        result = 1;
    } else if (arg0 & 0x03000242) {
        result = 2;
    } else if (arg0 & 0x48800400) {
        result = 4;
    } else if (arg0 & 0x04000000) {
        result = 8;
```
Note that arg0 probably has to be made a proper u32 for this to work as it should.

Reply:
s32 works just fine